server/rest/src/main/java/org/infinispan/rest/resources/CacheResource.java

```diff
@@ -0,0 +1,363 @@
+                  return responseBuilder.status(HttpResponseStatus.PRECONDITION_FAILED).build();
+               }
+               if (CacheOperationsHelper.ifUnmodifiedIsBeforeEntryModificationDate(ifUnmodifiedSince, lastMod)) {
+                  return responseBuilder.status(HttpResponseStatus.PRECONDITION_FAILED).build();
+               }
+               if (CacheOperationsHelper.ifModifiedIsAfterEntryModificationDate(ifModifiedSince, lastMod)) {
+                  return responseBuilder.status(HttpResponseStatus.NOT_MODIFIED).build();
+               }
+               Object value = ice.getValue();
+               MediaType configuredMediaType = restCacheManager.getValueConfiguredFormat(cacheName);
+               writeValue(value, requestedMediaType, configuredMediaType, responseBuilder, returnBody);
+
+               responseBuilder.status(HttpResponseStatus.OK)
+                     .lastModified(lastMod)
+                     .eTag(etag)
+                     .cacheControl(CacheOperationsHelper.calcCacheControl(expires))
+                     .expires(expires)
+                     .timeToLive(meta.lifespan())
+                     .maxIdle(meta.maxIdle());
+
+               List<String> extended = request.parameters().get("extended");
+               if (extended != null && extended.size() > 0 && CacheOperationsHelper.supportsExtendedHeaders(restServerConfiguration, extended.iterator().next())) {
+                  responseBuilder.clusterPrimaryOwner(restCacheManager.getPrimaryOwner(cacheName, key))
+                        .clusterNodeName(restCacheManager.getNodeName())
+                        .clusterServerAddress(restCacheManager.getServerAddress());
+               }
+            }
+         }
+         return responseBuilder.build();
+      } catch (CacheException cacheException) {
+         throw createResponseException(cacheException);
+      }
+   }
+
+   private void writeValue(Object value, MediaType requested, MediaType configuredMediaType, NettyRestResponse.Builder
+         responseBuilder, boolean returnBody) {
+      MediaType responseContentType;
+
+      if (!requested.matchesAll()) {
+         responseContentType = requested;
+      } else {
+         if (configuredMediaType == null) {
+            responseContentType = value instanceof byte[] ? MediaType.APPLICATION_OCTET_STREAM : MediaType.TEXT_PLAIN;
+         } else {
+            responseContentType = configuredMediaType;
+         }
+      }
+      responseBuilder.contentType(responseContentType);
+
+      if (returnBody) responseBuilder.entity(value);
+   }
+
+   private <V> String calcETAG(V value) {
+      return String.valueOf(hashFunc.hash(value));
+   }
+
+   private RestResponse putInCache(NettyRestResponse.Builder responseBuilder, boolean useAsync, AdvancedCache<
+         Object, Object> cache, Object key,
+                                   byte[] data, Long ttl, Long idleTime, Optional<Object> prevCond) {
```

Comment:
`prevCond` param shouldn't be optional, just handle if null.

Reply:
Good catch, I forgot to remove this Optional